### ISSUE TITLE
Support for explicit file extensions when using ESM

### ISSUE THREAD

**Issue by DCsunset** [rhys-vdw/ts-auto-guard#227]
When using ESM with `moduleResolution` set to `nodenext` in `tsconfig.json`, the generated guard.ts will fail to compile:

```
error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'.
```

To fix it, the generated guard.ts should append `.js` to all its imports.

**Comment by DCsunset** [comment]
I think we can probably check whether the module is set to ESM and conditionally assign the `.js` extension to `relativeOutPath`. Or we could add a cli option to specify the extension to add.

**Comment by rhys-vdw** [comment]
Sounds good. I've never messed with ESM, so happy to take your lead here. Shouldn't be too hard a change.

**Comment by p15martin** [comment]
Did this ever get resolved? I'm hitting the same issue

**Comment by rhys-vdw** [comment]
@p15martin This issue has been fixed by @DCsunset and is available in 4.2.0

https://github.com/rhys-vdw/ts-auto-guard/actions/runs/8041457915
